crates/uv-resolver/src/lock/mod.rs

```diff
@@ -2871,6 +2871,19 @@ impl PackageWire {
         requires_python: &RequiresPython,
         unambiguous_package_ids: &FxHashMap<PackageName, PackageId>,
     ) -> Result<Package, LockError> {
+        // Consistency check
+        if let Some(version) = &self.id.version {
+            for wheel in &self.wheels {
+                if version != &wheel.filename.version {
+                    return Err(LockError::from(LockErrorKind::InconsistentVersions {
+                        name: self.id.name,
+                    }));
```

Comment:
Can we show the versions in the error here? Like "packaged version {} does not match {} from wheel {}"?

Reply:
https://github.com/astral-sh/uv/pull/12249